producer.py

```diff
@@ -0,0 +1,27 @@
+import functools
+import redis
+
+
+class Producer:
+
+    def __init__(self, stream_name, host, port, db):
+        self.stream = stream_name
+        self._r = redis.Redis(host=host, port=port, db=db)
+
+    def event(self, action, data={}):
```

Comment:
data={} is a mutable default argument, which is really bad practice https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments. in this particular case it's not really an issue, but avoid it